src/main/starlark/builtins_bzl/common/java/java_binary_deploy_jar.bzl

```diff
@@ -276,12 +279,19 @@ def make_deploy_jars_rule(implementation):
             ),
             "_cc_toolchain": attr.label(default = "@" + cc_semantics.get_repo() + "//tools/cpp:current_cc_toolchain"),
             "_java_toolchain_type": attr.label(default = semantics.JAVA_TOOLCHAIN_TYPE),
-            "_java_runtime_toolchain_type": attr.label(default = semantics.JAVA_RUNTIME_TOOLCHAIN_TYPE),
             "_build_info_translator": attr.label(
                 default = semantics.BUILD_INFO_TRANSLATOR_LABEL,
             ),
-        },
+        } | (
+            {
+                "_java_runtime_toolchain_type": attr.label(
+                    default = semantics.JAVA_RUNTIME_TOOLCHAIN_TYPE,
+                ),
+            } if create_executable else {}
+        ),
         outputs = _implicit_outputs,
         fragments = ["java"],
-        toolchains = [semantics.JAVA_TOOLCHAIN, semantics.JAVA_RUNTIME_TOOLCHAIN] + cc_helper.use_cpp_toolchain(),
+        toolchains = [semantics.JAVA_TOOLCHAIN] + cc_helper.use_cpp_toolchain() + (
+            [semantics.JAVA_RUNTIME_TOOLCHAIN] if create_executable else []
```

Comment:
There's been discussion internally wrt Starlark code style and consensus is to avoid ternary `if`s, even for one-liners. 
For consistency, please extract this into a variable and conditionally add the runtime toolchain with a regular `if`.